src/component/button/PermalinkButton/PermalinkButton.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react';
+
+import SimpleButton from '@terrestris/react-geo/dist/Button/SimpleButton/SimpleButton';
+import Window from '@terrestris/react-geo/dist/Window/Window';
+
+import { TooltipPlacement } from 'antd/lib/tooltip';
+import Button, { ButtonProps } from 'antd/lib/button';
+import { message } from 'antd';
+
+import { PermalinkUtil } from '../../../util/PermalinkUtil/PermalinkUtil';
+
+import './PermalinkButton.css';
+
+
+interface DefaultPermalinkButtonProps {
+  type: 'default' | 'primary' | 'ghost' | 'dashed' | 'danger' | 'link';
+  shape: 'circle' | 'round';
+  iconName: string;
+}
+
+interface BaseProps {
+  map: any;
+  tooltip: string;
+  tooltipPlacement: TooltipPlacement;
+  t: (arg: string) => string;
+  getLink?: () => string;
+}
+
+interface PermalinkButtonState {
+  winVisible: boolean;
+}
+
+export type PermalinkButtonProps = BaseProps & Partial<DefaultPermalinkButtonProps> & ButtonProps;
+
+/**
+ * Class representing the PermalinkButton.
+ *
+ * @class PermalinkButton
+ * @extends React.Component
+ */
+export default class PermalinkButton extends React.Component<PermalinkButtonProps, PermalinkButtonState> {
+
+
+  /**
+   * The default properties.
+   */
+  public static defaultProps: DefaultPermalinkButtonProps = {
+    type: 'primary',
+    shape: 'circle',
+    iconName: 'fas fa-link'
+  };
+
+  /**
+   * Create the PermalinkButton. Opens window which creates a permalink.
+   *
+   * @constructs PermalinkButton
+   */
+  constructor(props: PermalinkButtonProps) {
+    super(props);
+    this.state = {
+      winVisible: false
+    };
+  }
+
+  /**
+   * Inverts visibility of the permalink window.
+   */
+  changeVisibility() {
+    this.setState({
+      winVisible: !this.state.winVisible
+    });
+  };
+
+  /**
+   * Copy the permalink to clipboard
+   */
+  copyToClipBoard() {
+    const {
+      t
+    } = this.props;
+
+    const linkInput: HTMLInputElement = document.querySelector(
+      '.permalink input');
+
+    if (linkInput) {
+      linkInput.select();
+      linkInput.setSelectionRange(0, 99999);
+      document.execCommand('copy');
+      message.info(t('Permalink.copyiedToClipboard'));
+    }
+  }
```

Comment:
This should either moved to the `base-util` or a third party module (e.g. `copy-to-clipboard`) should be utilized.